interfaces/builtin/registry.go

```diff
@@ -70,10 +70,8 @@ func (iface *registryInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
 	}
 
 	role, ok := plug.Attrs["role"].(string)
-	if !ok {
-		// defaults to manager
-		role = "manager"
-		plug.Attrs["role"] = role
+	if !ok || role == "" {
+		return fmt.Errorf(`registry plug must have a valid "role" attribute`)
```

Comment:
should we actually unify this check with the one below? Also thinking further I wonder if actually we should allow this to be left out without default maybe or have a third value?

* manager:  snap can access  the view and have change-registry etc hooks called for the view
* obsever: snap can access the view and can have registry-changed hooks invoked for the view
* `emtpy`:  snap can only access the view

maybe something to rediscuss?

Reply:
What would be the use case? If a snap can read the view, then the same level of "access" is implied and it can just have a no-op hook if it doesn't want to be notified.